src/aleph/handlers/content/content_handler.py

```diff
@@ -47,6 +47,19 @@ async def process(self, session: DbSession, messages: List[MessageDb]) -> None:
         """
         pass
 
+    async def check_balance(self, session: DbSession, message: MessageDb) -> None:
+        """
+        Check user's balance and cost.
+
+        This function should verify whether the user has sufficient balance to perform
+        the requested action.
+        It's raise InsufficientBalanceException
```

Comment:
```suggestion
        Checks whether the user has enough Aleph tokens to process the message.

        Raises InsufficientBalanceException if the balance of the user is too low.
```

Reply:
Also, I thought about this, but it might actually be better to define the function as `compute_cost(session: DbSession, message: MessageDb) -> Decimal` and then mutualize all the logic to check the current costs in the `MessageHandler` class. What do you think?